addons/mail/static/tests/discuss_app/discuss_tests.js

```diff
@@ -48,62 +49,79 @@ QUnit.test("sanity check", async (assert) => {
+    await contains("input.o-mail-Discuss-threadName", { value: "special" });
+    await def;
     assert.verifySteps(["/web/dataset/call_kw/discuss.channel/channel_rename"]);
 });
 
-QUnit.test("can change the thread description of #general", async (assert) => {
+QUnit.test("can active change thread from messaging menu", async (assert) => {
+    const pyEnv = await startServer();
+    const [, teamId] = pyEnv["discuss.channel"].create([
+        { name: "general", channel_type: "channel" },
+        { name: "team", channel_type: "channel" },
+    ]);
+    const { openDiscuss } = await start();
+    await openDiscuss(teamId);
+    await contains(".o-mail-DiscussCategoryItem", { text: "general" });
+    await contains(".o-mail-DiscussCategoryItem.o-active", { text: "team" });
+    await click("[aria-label='Messages']");
+    await click(".o-mail-NotificationItem:contains(general)");
+    await contains(".o-mail-DiscussCategoryItem.o-active", { text: "general" });
+    await contains(".o-mail-DiscussCategoryItem", { text: "team" });
+});
+
+QUnit.test("can change the thread description of #general [REQUIRE FOCUS]", async (assert) => {
     const pyEnv = await startServer();
     const channelId = pyEnv["discuss.channel"].create({
         name: "general",
         channel_type: "channel",
         description: "General announcements...",
+        create_uid: pyEnv.currentPartnerId,
     });
+    const def = new Deferred();
     const { openDiscuss } = await start({
         mockRPC(route, params) {
             if (route === "/web/dataset/call_kw/discuss.channel/channel_change_description") {
                 assert.step(route);
+                def.resolve();
             }
         },
     });
-    await openDiscuss(channelId);
-    assert.containsOnce($, "input.o-mail-Discuss-threadDescription");
-    const $description = $("input.o-mail-Discuss-threadDescription");
-
-    click($description).then(() => {});
-    assert.strictEqual($description.val(), "General announcements...");
-    await editInput(
-        document.body,
-        "input.o-mail-Discuss-threadDescription",
-        "I want a burger today!"
-    );
-    await triggerEvent(document.body, "input.o-mail-Discuss-threadDescription", "keydown", {
-        key: "Enter",
+    openDiscuss(channelId);
+    await contains(".o-mail-Composer-input:focus");
+    await contains("input.o-mail-Discuss-threadDescription", {
+        value: "General announcements...",
     });
-    assert.strictEqual($description.val(), "I want a burger today!");
+    await insertText("input.o-mail-Discuss-threadDescription:enabled", "I want a burger today!", {
+        replace: true,
+    });
+    triggerHotkey("Enter");
+    await contains("input.o-mail-Discuss-threadDescription", {
+        value: "I want a burger today!",
+    });
+    await def;
     assert.verifySteps(["/web/dataset/call_kw/discuss.channel/channel_change_description"]);
```

Comment:
note: if we're bothering to check the rpc, we could at least check it receives the correct value

Reply:
deferred added